solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -845,27 +849,30 @@ contract LiquidityProtection is TokenHandler, ContractRegistryClient, Reentrancy
         IERC20Token _reserveToken,
         uint256 _poolAmount,
         uint256 _reserveAmount,
-        Fraction memory _addRate,
-        Fraction memory _removeRate,
+        PackedRates memory _packedRates,
         uint256 _addTimestamp,
         uint256 _removeTimestamp)
         internal view returns (uint256)
     {
-        // get the adjusted amount of reserve tokens based on the exposure and rate changes
-        uint256 outputAmount = adjustedAmount(_poolToken, _reserveToken, _poolAmount, _addRate, _removeRate);
+        // get the rate between the reserves upon adding liquidity and now
+        Fraction memory addSpotRate = Fraction({n: _packedRates.addSpotRateN, d: _packedRates.addSpotRateD});
+        Fraction memory removeSpotRate = Fraction({n: _packedRates.removeSpotRateN, d: _packedRates.removeSpotRateD});
+        Fraction memory removeAverageRate = Fraction({n: _packedRates.removeAverageRateN, d: _packedRates.removeAverageRateD});
+
+        // calculate the entitled amount of reserve tokens before compensation
+        uint256 entitlement = entitledAmount(_poolToken, _reserveToken, _poolAmount, addSpotRate, removeSpotRate);
```

Comment:
output amount instead of entitlement? totalAmount? reserveAmountPlusFees?

Reply:
A local variable by the name `totalAmount` is used in various places, so we cannot use this one.
Please specify what name you want for the variable and what name you want for the function.

Reply:
reserveAmountPlusFees then

Reply:
Named it `protectedAmountPlusFee`.